overlord/snapstate/aliases.go

```diff
@@ -0,0 +1,338 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snapstate
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/tomb.v2"
+
+	"github.com/snapcore/snapd/i18n/dumb"
+	"github.com/snapcore/snapd/overlord/snapstate/backend"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+)
+
+type AliasState struct {
+	// Enabled tracks for which snap the alias has been enabled, "" if none
+	Enabled string `json:"enabled,omitempty"`
+	// Auto is true if the alias has been auto-enabled
+	// TODO: Auto bool `json:"auto,omitempy"`
+	// Disabled tracks the possibly many snap for which the alias has
+	// been forcefully disabled
+	// TODO: Disabled []string `json:"disabled,omitempty"`
+}
+
+type aliasStateMap map[string]*AliasState
+
+func getAliases(st *state.State) (aliasStateMap, error) {
+	var aliasStates map[string]*AliasState
+	err := st.Get("aliases", &aliasStates)
+	if err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+	return aliasStates, nil
+}
+
+func (m aliasStateMap) get(alias string) AliasState {
```

Comment:
This type feels a bit atypical. The result of get is not a pointer, but the argument of set is a pointer.. get is basically just doing the usual map-getting thing, and set seems to just do the set as well, but deletes on nil. The code would probably be more straightforward and expected if we just kept using the map itself as usual.